src/v/redpanda/application.cc

```diff
@@ -713,16 +716,45 @@ void application::wire_up_redpanda_services() {
           std::ref(metadata_cache),
           std::ref(partition_manager));
         coprocessing->start().get();
+    }
 
+    if (v8_enabled()) {
+        syschecks::systemd_message("Creating v8::api").get();
+        const auto& cfg = config::shard_local_cfg();
+        construct_single_service(
+          _v8_engine,
+          ss::engine().alien(),
+          cfg.core_for_executor,
+          cfg.executor_queue_size);
+        _v8_engine->start().get();
+    }
+
+    if (coproc_enabled() || v8_enabled()) {
+        syschecks::systemd_message(
+          "Creating event listener for coproc internal topic")
+          .get();
         construct_single_service(_coproc_event_listener);
 
-        _async_event_handler
-          = std::make_unique<coproc::wasm::async_event_handler>(
-            _coproc_event_listener->get_abort_source(),
-            coprocessing->get_pacemaker());
+        if (coproc_enabled()) {
+            _async_event_handler
+              = std::make_unique<coproc::wasm::async_event_handler>(
+                _coproc_event_listener->get_abort_source(),
+                coprocessing->get_pacemaker());
+
+            _coproc_event_listener->register_handler(
```

Comment:
Also if the event listeners don't have shared state there is no reason for them to exist as unique pointers. The event listener can just hold private references to them internally, sending updates to them as they arrive

Reply:
Did you about _coproc_event_listener or about handlers?

_coproc_event_listener is pointer because we use it for single service for redpanda. 
If you wrote about handlers, I need to use optional or poiunter, because handler for coproc needs abourt_source and pacemaker on construct methods. But I may not exist it coproc disabled